src/main/webapp/app/exam/manage/suspicious-behavior/suspicious-behavior.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, OnInit } from '@angular/core';
+import { Exercise } from 'app/entities/exercise.model';
+import { SuspiciousExamSessions } from 'app/entities/exam-session.model';
+import { SuspiciousSessionsService } from 'app/exam/manage/suspicious-behavior/suspicious-sessions.service';
+import { ActivatedRoute, Router } from '@angular/router';
+import { PlagiarismCasesService } from 'app/course/plagiarism-cases/shared/plagiarism-cases.service';
+import { ExamManagementService } from 'app/exam/manage/exam-management.service';
+import { PlagiarismResultsService } from 'app/course/plagiarism-cases/shared/plagiarism-results.service';
+
+@Component({
+    selector: 'jhi-suspicious-behavior',
+    templateUrl: './suspicious-behavior.component.html',
+})
+export class SuspiciousBehaviorComponent implements OnInit {
+    exercises: Exercise[] = [];
+    plagiarismCasesPerExercise: Map<Exercise, number> = new Map<Exercise, number>();
+    plagiarismResultsPerExercise: Map<Exercise, number> = new Map<Exercise, number>();
+    anyPlagiarismCases = false;
+    suspiciousSessions: SuspiciousExamSessions[] = [];
+    examId: number;
+    courseId: number;
+
+    constructor(
+        private suspiciousSessionsService: SuspiciousSessionsService,
+        private activatedRoute: ActivatedRoute,
+        private plagiarismCasesService: PlagiarismCasesService,
+        private examService: ExamManagementService,
+        private plagiarismResultsService: PlagiarismResultsService,
+        private router: Router,
+    ) {}
+
+    ngOnInit(): void {
+        this.examId = Number(this.activatedRoute.snapshot.paramMap.get('examId'));
+        this.courseId = Number(this.activatedRoute.snapshot.paramMap.get('courseId'));
+        this.suspiciousSessionsService.getSuspiciousSessions(this.courseId, this.examId).subscribe((res) => {
+            this.suspiciousSessions = res;
+        });
+        this.examService.getExercisesWithPotentialPlagiarismForExam(this.courseId, this.examId).subscribe((res) => {
+            this.exercises = res;
+            this.retrievePlagiarismCases();
+        });
+    }
+
+    retrievePlagiarismCases = () => {
+        this.exercises.forEach((exercise) => {
+            this.plagiarismCasesService.getNumberOfPlagiarismCasesForExercise(exercise).subscribe((res) => {
+                this.plagiarismCasesPerExercise.computeIfAbsent(exercise, () => res);
+                if (res > 0) this.anyPlagiarismCases = true;
```

Comment:
Is this equivalent in this context? If so, I would write it like this:
```suggestion
                this.anyPlagiarismCases = res > 0;
```

Reply:
unfortunately not, because if we apply your logic it will always depend on the last value omitted  from the observable which is not true. e.g exercise 1 has plagiarism cases and exercise 3 (assuming 3 exercises) hasn't, it should be true but would be false with your suggestion.

Reply:
Ok 👍